File: DATN-Fashion-Shop-BE/src/main/java/com/example/DATN_Fashion_Shop_BE/model/ProductMedia.java
package com.example.DATN_Fashion_Shop_BE.model;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.envers.Audited;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "product_media")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ProductMedia extends BaseEntity{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "media_url", nullable = false)
    private String mediaUrl;

    @Column(name = "media_type", nullable = false)
    private String mediaType;

    @Column(name = "sort_order")
    private Integer sortOrder;

    @Column(name = "model_height")
    private Integer modelHeight; // Chiều cao của mẫu sản phẩm trong hình ảnh

    @ManyToOne
    @JoinColumn(name = "color_value_id", referencedColumnName = "id")
    private AttributeValue colorValue; // Liên kết với bảng Attributes_Values

    @ManyToOne
    @JoinColumn(name = "product_id", nullable = false)
    private Product product; // Liên kết với bảng Products

    @ManyToMany(cascade = {CascadeType.MERGE})
    @JoinTable(
            name = "media_variants", // Bảng trung gian
            joinColumns = @JoinColumn(name = "product_media_id"), // FK tới ProductMedia
            inverseJoinColumns = @JoinColumn(name = "product_variant_id") // FK tới ProductVariant
    )
    private List<ProductVariant> productVariants = new ArrayList<>();
}

File: FE/DATN-Fashion-Shop-FE/src/app/models/Product/ProductMedia.ts
import { BaseEntity} from '../BaseEntity';
import { AttributeValue} from './AttributeValue';
import { Product} from './product';
import { ProductVariant} from './ProductVariant';

export class ProductMedia extends BaseEntity {
  id: number;
  mediaUrl: string;
  mediaType: string;
  sortOrder?: number;
  modelHeight?: number;
  colorValue?: AttributeValue;
  product: Product;
  productVariants: ProductVariant[];

  constructor(data?: Partial<ProductMedia>) {
    super(data);
    this.id = data?.id || 0;
    this.mediaUrl = data?.mediaUrl || '';
    this.mediaType = data?.mediaType || '';
    this.sortOrder = data?.sortOrder ?? undefined;
    this.modelHeight = data?.modelHeight ?? undefined;
    this.colorValue = data?.colorValue ? new AttributeValue(data.colorValue) : undefined;
    this.product = data?.product ? new Product(data.product) : new Product();
    this.productVariants = data?.productVariants?.map(pv => new ProductVariant(pv)) || [];
  }
}
